testing/src/org/immutables/check/IterableChecker.java

```diff
@@ -70,8 +70,8 @@ public final void isOf(T... elements) {
     verifyUsingMatcher(Matchers.contains(elements));
   }
 
-  public void hasContentInAnyOrder(Iterable<?> elements) {
-    verifyUsingMatcher((Matcher<? super Z>) Matchers.containsInAnyOrder(ImmutableList.copyOf(elements)));
+  public void hasContentInAnyOrder(Iterable<T> elements) {
+    verifyUsingMatcher(Matchers.containsInAnyOrder((T[]) Iterables.toArray(elements, Object.class)));
```

Comment:
cannot verify right now, is there any warnings to suppress `@SuppressWarnings("unchecked")` or can cast `(T[])` be removed? just asking. Thank you or PR!

Reply:
@elucash you're right, I'd removed redundant cast